app/controllers/users_controller.rb

```diff
@@ -85,11 +84,24 @@ def show
       render_404
     else
       respond_to do |format|
-        format.html { render layout: 'no_menu' }
+        format.html do
+          show_respond_html(events)
+        end
       end
     end
   end
 
+  def show_respond_html(events)
+    @events_by_day = events.group_by { |e| e.event_datetime.to_date }
+    @journals_by_id = Journal
+      .includes(:data, :customizable_journals, :attachable_journals, :bcf_comment)
+      .find(events.pluck(:event_id))
+      .then { |journals| ::API::V3::Activities::ActivityEagerLoadingWrapper.wrap(journals) }
+      .index_by(&:id)
+
+    render layout: 'no_menu'
+  end
```

Comment:
This is almost identical to the code in the `activities_controller`. Please extract this into a shared method.

Reply:
Yes, almost identical. I was not sure about how to extract it. A shared method is not the right abstraction IMHO. event and journal should be tied together at a higher level.

I'll try to refactor it in a better way to remove this logic duplication.

Reply:
Should be ok now with latest commit.